File: repair-management-api/Models/Customer.cs
namespace RepairManagementApi.Models;

public class Customer
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string FullName { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string? Email { get; set; } = string.Empty;
    public string? Address { get; set; }
    public Guid BranchId { get; set;}
    public Branch Branch { get; set; }
    public DateTime CreatedAtUtc { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAtUtc { get; set; } = DateTime.UtcNow;

}

File: repair-management-app/src/types/customer.ts
export type CustomerListItem = {
  id: string;
  fullName: string;
  phone: string;
  email: string | null;
  address: string | null;
  branchId: string;
  createdAtUtc: string;
  updatedAtUtc: string;
};

export type CreateCustomerRequest = {
  fullName: string;
  phone: string;
  email: string | null;
  address: string | null;
  branchId: string;
};
